packages/client/src/features/change-bounds/resize-tool.ts

```diff
@@ -0,0 +1,157 @@
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import { Action, ChangeBoundsOperation, Dimension, Point } from '@eclipse-glsp/protocol';
+import { inject, injectable, optional } from 'inversify';
+import {
+    KeyListener,
+    KeyTool,
+    SModelElement,
+    isBoundsAware,
+    isSelectable,
+    BoundsAware,
+    SParentElement,
+    ISnapper,
+    Selectable,
+    SModelRoot
+} from 'sprotty';
+import { TYPES } from '../../base/types';
+import { isResizable, Resizable } from '../change-bounds/model';
+import { matchesKeystroke } from 'sprotty/lib/utils/keyboard';
+import { GLSPTool } from '../../base/tool-manager/glsp-tool-manager';
+import { toElementAndBounds } from '../../utils/smodel-util';
+import { GridSnapper } from './snap';
+import { isValidMove, isValidSize, minHeight, minWidth } from '../../utils/layout-utils';
+import { IMovementRestrictor } from './movement-restrictor';
+
+@injectable()
+export class ResizeTool implements GLSPTool {
+    static ID = 'glsp.resize-keyboard';
+
+    isEditTool = true;
+
+    @inject(KeyTool) protected readonly keytool: KeyTool;
+    @inject(TYPES.IMovementRestrictor) @optional() readonly movementRestrictor?: IMovementRestrictor;
+    @inject(TYPES.ISnapper) @optional() readonly snapper?: ISnapper;
+
+    protected resizeKeyListener: ResizeKeyListener;
+
+    get id(): string {
+        return ResizeTool.ID;
+    }
+
+    enable(): void {
+        this.resizeKeyListener = new ResizeKeyListener(this);
+        this.keytool.register(this.resizeKeyListener);
+    }
+
+    disable(): void {
+        this.keytool.deregister(this.resizeKeyListener);
+    }
+}
+@injectable()
+export class ResizeKeyListener extends KeyListener {
+    protected grid = { x: 20, y: 20 };
+    protected isEditMode = false;
+
+    constructor(protected readonly tool: ResizeTool) {
+        super();
+
+        if (this.tool.snapper instanceof GridSnapper) {
+            this.grid = this.tool.snapper?.grid;
+        }
+    }
+
+    override keyDown(element: SModelElement, event: KeyboardEvent): Action[] {
+        const actions: Action[] = [];
+
+        if (this.isEditMode && matchesKeystroke(event, 'Escape')) {
+            this.isEditMode = false;
+        }
+
+        if (this.getSelectedElements(element.root).length > 0) {
+            if (!this.isEditMode && matchesKeystroke(event, 'KeyR', 'alt')) {
```

Comment:
>  Unfortunately Alt+R is already used in the Theia integration for Reveal in Explorer which is why this keybinding will not get triggered properly. I think we need to find a good alternative for it.

1. One possibility would be to prevent the event from coming to Theia. Because especially when the element is currently focused/selected, only the associated actions should be available.
2. Replace ALT+R with e.g. one of the function keys (F1...F12).

Reply:
Just to follow up on this: the function keys are also quite risky in a browser behavior as at least F1 (Help) and F5 (Reload). So we definitely need to find a valid keybinding before we can merge this.